FWCore/Framework/src/PrincipalGetAdapter.cc

```diff
@@ -79,6 +72,12 @@ namespace edm {
                                         << "The index of the token was " << token.index() << ".\n";
   }
 
+  void principal_get_adapter_detail::throwGetManyByType() {
+    throw Exception(errors::LogicError)
```

Comment:
Or `errors::UnimplementedFeature`? I guess I'm fine with either in the end.

Reply:
I was debating between those two myself while I was implementing this PR. My final thought was that UnimplementedFeature is incorrect. It implies we did not implement it yet, but maybe will someday. That is different than we implemented it and then removed it. I would argue LogicError is correct although not very informative. We could add a new category, either "DeprecatedFeature" or "RemovedFeature". I'm not sure it is worth a new category though given the number of things that might be impacted by adding a new exit code. Not feeling strongly about this. I'll change it if you like.

This is also going to get entirely deleted soon when I completely remove getManyByType.

Reply:
Good point, let's stay with `LogicError`.